File: swing/src/main/java/wannabe/swing/renderer/FilledSquare.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.Projected;

public class FilledSquare extends SwingRenderer {
  @Override public void draw(Graphics g, Projected r) {
    g.fillRect(r.left, r.top, r.size, r.size);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/FilledSquare.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class FilledSquare : SwingRenderer() {
  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    g.fillRect(p.left, p.top, p.size, p.size)
  }
}
